freebites/product
# 27 post history

# Description



1. Created the functionality and the frontend work for the delete icon, including the modal that pops up when the trash icon is pressed.
2. Corrected the functionality of the "Your Post" to only show the posts that you post yourself.
3. Created the functionality of the "Post History" to show the posts that you post yourself.

## Issues

Resolve #27 




## Screenshots

 
<img src="https://github.com/freebites/product/assets/158139019/3b14c69f-77e8-4a0c-ae4d-37109a396c6e" width="200" />
<img src="https://github.com/freebites/product/assets/158139019/4577c663-e38a-461a-919f-0097e5c65c4e" width="200" />
<img src="https://github.com/freebites/product/assets/158139019/d8c6a487-7c67-4e7c-b158-555274deb9ea" width="200" />
<img src="https://github.com/freebites/product/assets/158139019/90fa38e8-ab9b-4fc6-999f-3331df560a20" width="200" />


## Test



1. Create a new post
2. Check "Your Post" to see if the post shows up
3. Check "History" to see if the post shows up
4. Check the post to see if the trash icon is on the post
5. Click on the trash icon to see if the delete modal shows up
6. Hit "Yes" on the delete modal and see if the post automatically deletes from the page
7. Check both "Your Post" and "History" to see if it is deleted there

## Possible Downsides



1. If you delete the post in "History", it still does not disappear from the Home, which could cause some errors if a user does try to open up the post after it's been deleted. However, refreshing the page does fix the problem.
2. The delete modal backdrop when it is open does prevent the user from clicking on other posts, but it doesn't make the modal disappear. 
3. We implemented two new props into the "HomePost" document: setRefreshing and fetchData. This could cause some complications in the future if someone does want to use "HomePost".

## Additional Documentations

 N/A
